### PR TITLE
Epoll fixes in safeposix, updated net_calls.rs

### PR DESCRIPTION
**Pull request by justyoyo123**

### CODE REVIEW

File: src/safeposix/syscalls/net_calls.rs [hunk 1]
```diff
@@ -1567,27 +1566,39 @@ impl Cage {
...
                     }
                     poll_fds_vec.push(structpoll);
+                  num_events += 1;
                 }
-
+                
                 let poll_fds_slice = &mut poll_fds_vec[..];
                 Self::poll_syscall(&self, poll_fds_slice, timeout);
-                let mut count_changed: i32 = 0;
+                let mut count = 0;
```
Comment (TristanB22) [inline src/safeposix/syscalls/net_calls.rs:1574]: should we define this to be a u32 or i32?

**Review by TristanB22 — COMMENTED** [review]
looks pretty good to me. I like the changes

File: src/safeposix/syscalls/net_calls.rs (hunk 1 above)
Comment (justyoyo123) [reply src/safeposix/syscalls/net_calls.rs:1574]: we let the compiler decide what type it should be and cast it to i32 at the end (line 1601)

File: src/safeposix/syscalls/net_calls.rs [hunk 2]
```diff
@@ -1547,7 +1546,7 @@ impl Cage {
                 }
```
Comment (Devansh-Patel) [inline src/safeposix/syscalls/net_calls.rs:1546]: On line 1544, it should be `if maxevents < 0 `

File: src/safeposix/syscalls/net_calls.rs [hunk 3]
```diff
@@ -1567,27 +1566,39 @@ impl Cage {
...
                     poll_fds_vec.push(structpoll);
+                  num_events += 1;
                 }
-
+                
                 let poll_fds_slice = &mut poll_fds_vec[..];
                 Self::poll_syscall(&self, poll_fds_slice, timeout);
-                let mut count_changed: i32 = 0;
+                let mut count = 0;
+                let mut end_idx: i32 = maxevents;
```
Comment (Devansh-Patel) [inline src/safeposix/syscalls/net_calls.rs:1575]: You could directly cast maxevents as usize here.

File: src/safeposix/syscalls/net_calls.rs [hunk 4]
```diff
@@ -1567,27 +1566,39 @@ impl Cage {
...
+                if num_events < maxevents {
```
Comment (Devansh-Patel) [inline src/safeposix/syscalls/net_calls.rs:1576]: Here, you could us min to set the value of end_idx in one line.

File: src/safeposix/syscalls/net_calls.rs [hunk 5]
```diff
@@ -1567,27 +1566,39 @@ impl Cage {
                         structpoll.events |= POLLERR;
                     }
                     poll_fds_vec.push(structpoll);
+                  num_events += 1;
                 }
-
+
```
Comment (Devansh-Patel) [inline src/safeposix/syscalls/net_calls.rs:1571]: There are some white-spaces on this line.

File: src/safeposix/syscalls/net_calls.rs [hunk 6]
```diff
@@ -1567,27 +1565,34 @@ impl Cage {
...
-
+                  num_events += 1;
+                } 
                 let poll_fds_slice = &mut poll_fds_vec[..];
                 Self::poll_syscall(&self, poll_fds_slice, timeout);
-                let mut count_changed: i32 = 0;
-
-                for (count, result) in poll_fds_slice[..maxevents as usize].iter().enumerate() {
+                let mut count = 0;
+                let end_idx: usize = core::cmp::min(num_events as usize, maxevents as usize);
```
Comment (rennergade) [inline src/safeposix/syscalls/net_calls.rs:1573]: we don't use outside imports directly in safeposix, instead use interface::rust_min

Comment (rennergade) [reply src/safeposix/syscalls/net_calls.rs:1573]: I also think num_events is already usize so you dont have to cast?

**Review by rennergade — APPROVED** [review]
Cool. This is approved now.

**Pull request merged**